discovery/kubernetes/client_metrics.go

```diff
@@ -251,31 +252,51 @@ func (f *clientGoWorkqueueMetricsProvider) NewDepthMetric(name string) workqueue
 func (f *clientGoWorkqueueMetricsProvider) NewAddsMetric(name string) workqueue.CounterMetric {
 	return clientGoWorkqueueAddsMetricVec.WithLabelValues(name)
 }
-func (f *clientGoWorkqueueMetricsProvider) NewLatencyMetric(name string) workqueue.SummaryMetric {
+func (f *clientGoWorkqueueMetricsProvider) NewLatencyMetric(name string) workqueue.HistogramMetric {
 	metric := clientGoWorkqueueLatencyMetricVec.WithLabelValues(name)
 	// Convert microseconds to seconds for consistency across metrics.
 	return prometheus.ObserverFunc(func(v float64) {
 		metric.Observe(v / 1e6)
 	})
 }
-func (f *clientGoWorkqueueMetricsProvider) NewWorkDurationMetric(name string) workqueue.SummaryMetric {
-	metric := clientGoWorkqueueWorkDurationMetricVec.WithLabelValues(name)
-	// Convert microseconds to seconds for consistency across metrics.
-	return prometheus.ObserverFunc(func(v float64) {
-		metric.Observe(v / 1e6)
-	})
+func (f *clientGoWorkqueueMetricsProvider) NewWorkDurationMetric(name string) workqueue.HistogramMetric {
+	return clientGoWorkqueueWorkDurationMetricVec.WithLabelValues(name)
 }
 func (f *clientGoWorkqueueMetricsProvider) NewUnfinishedWorkSecondsMetric(name string) workqueue.SettableGaugeMetric {
 	return clientGoWorkqueueUnfinishedWorkSecondsMetricVec.WithLabelValues(name)
 }
-func (f *clientGoWorkqueueMetricsProvider) NewLongestRunningProcessorMicrosecondsMetric(name string) workqueue.SettableGaugeMetric {
+func (f *clientGoWorkqueueMetricsProvider) NewLongestRunningProcessorSecondsMetric(name string) workqueue.SettableGaugeMetric {
 	metric := clientGoWorkqueueLongestRunningProcessorMetricVec.WithLabelValues(name)
 	return gaugeSetFunc(func(v float64) {
 		metric.Set(v / 1e6)
 	})
-
 }
 func (clientGoWorkqueueMetricsProvider) NewRetriesMetric(name string) workqueue.CounterMetric {
 	// Retries are not used so the metric is omitted.
 	return noopMetric{}
 }
+func (clientGoWorkqueueMetricsProvider) NewDeprecatedDepthMetric(name string) workqueue.GaugeMetric {
+	return clientGoWorkqueueDepthMetricVec.WithLabelValues(name)
```

Comment:
Shouldn't we return `noopMetric{}` here as well as for the other deprecated metrics?

Reply:
I thought it would be a safer bet to keep the implementation as-is. Since it's many of the previously implemented methods in k8s `1.13` that got deprecated in `k8s 1.14`.

I can change it to `noopMetric{}` if you think that's more appropriate.

Reply:
For gauges, it shouldn't matter much (although it is extra work) but the counters and summaries will double counted.

Reply:
Makes sense. I’ll make it noopMetric for all the deprecated methods then.